runtime-builder/tests/app/server.go

```diff
@@ -0,0 +1,258 @@
+)
+
+type appHandler func(http.ResponseWriter, *http.Request) error
+
+func (h appHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if err := h(w, r); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+var (
+	ctx       = context.Background()
+	projectID string
+)
+
+func main() {
+	if metadata.OnGCE() {
+		var err error
+		if projectID, err = metadata.ProjectID(); err != nil {
+			log.Fatal("getting project ID on GCE:", err)
+		}
+	} else {
+		projectID = os.Getenv("PROJECT_ID")
+	}
+	http.HandleFunc("/", mainHandler)
+	http.HandleFunc("/_ah/health", healthCheckHandler)
+	http.HandleFunc("/version", versionHandler)
+	http.Handle("/lookup_host", appHandler(lookupHostHandler))
+	http.Handle("/logging", appHandler(loggingHandler))
+	http.Handle("/logging_standard", appHandler(standardLoggingHandler))
+	http.Handle("/logging_custom", appHandler(customLoggingHandler))
+	http.Handle("/monitoring", appHandler(monitoringHandler))
+	http.Handle("/exception", appHandler(exceptionHandler))
+	http.Handle("/custom", appHandler(customHandler))
+	log.Print("Listening on port 8080")
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+func mainHandler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.NotFound(w, r)
+		return
+	}
+	fmt.Fprint(w, "Hello World!")
+}
+
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprint(w, "ok")
+}
+
+func versionHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Go version=%s\nGOARCH=%s\nGOOS=%s\n", runtime.Version(), runtime.GOARCH, runtime.GOOS)
+}
+
+func lookupHostHandler(w http.ResponseWriter, r *http.Request) error {
+	addrs, err := net.LookupHost(r.Host)
+	if err != nil {
+		return fmt.Errorf("error lookup host: %v", err)
+	}
+	fmt.Fprint(w, strings.Join(addrs, "\n"))
+	return nil
+}
+
+func loggingHandler(w http.ResponseWriter, r *http.Request) error {
+	c, err := logging.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("failed to create logging client: %v", err)
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	var b struct {
+		LogName string `json:"log_name"`
+		Token   string `json:"token"`
+	}
+	if err := decoder.Decode(&b); err != nil {
+		return fmt.Errorf("decode request body: %v", err)
+	}
+	defer r.Body.Close()
+
+	lg := c.Logger(b.LogName)
+	lg.LogSync(ctx, logging.Entry{Payload: b.Token})
+	return nil
+}
+
+func standardLoggingHandler(w http.ResponseWriter, r *http.Request) error {
+	c, err := logging.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("failed to create logging client: %v", err)
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	var b struct {
+		Token string `json:"token"`
+		Level string `json:"level"`
+	}
+	if err := decoder.Decode(&b); err != nil {
+		return fmt.Errorf("decode request body: %v", err)
+	}
+	defer r.Body.Close()
+
+	lg := c.Logger("appengine.googleapis.com/stderr")
+	slg := lg.StandardLogger(logging.ParseSeverity(b.Level))
```

Comment:
Curious.  The test description at

https://github.com/GoogleCloudPlatform/runtimes-common/tree/master/integration_tests#standard-logging

mentions "..it will log the token at the provided level through its standard logging module".  Does it have such integration in Go's log package?  I wonder though how level translate to the log package since it has no notion of level.

If so, should we be using that approach instead?

It also mentions "...the sample application will return back to the test driver appengine.googleapis.com%2Fstderr".  Shouldn't this handler write this to the response?